flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/ProcessFunction.java

```diff
@@ -24,32 +24,24 @@
 import org.apache.flink.streaming.api.TimerService;
 import org.apache.flink.util.Collector;
 
-import java.io.Serializable;
-
 /**
- * Base interface for timely flatMap functions. FlatMap functions take elements and transform them,
- * into zero, one, or more elements. Typical applications can be splitting elements, or unnesting lists
- * and arrays.
- *
- * <p>A {@code TimelyFlatMapFunction} can, in addition to the functionality of a normal
- * {@link org.apache.flink.api.common.functions.FlatMapFunction}, also set timers and react
- * to them firing.
+ * A function that processes elements of a stream.
  *
- * <pre>{@code
- * DataStream<X> input = ...;
- *
- * DataStream<Y> result = input.flatMap(new MyTimelyFlatMapFunction());
- * }</pre>
+ * <p>The function will be called for every element in the input stream and can produce
+ * zero or more output. The function can also query the time and set timers. When
+ * reacting to the firing of set timers the function can emit yet more elements.
  *
```

Comment:
<p>The function will be called for every element in the input stream and can produce
 * zero or more output elements. Contrary to the {@link org.apache.flink.api.common.functions.FlatMapFunction},
 * this function can also query the time (both event and processing) and set timers,
 * through the provided {@link Context}. When reacting to the firing of set timers the function can
 * directly emit a result, and/or register a timer that will trigger an action in the future.